File: lab4/Java/Gerente.java
public class Gerente extends Funcionario implements InterfaceGerente {
    protected int qtdEquipes;

    public Gerente(String nomeFuncionario, float salarioFuncionario, int qtdEquipes) {
        super(nomeFuncionario, salarioFuncionario);
        this.qtdEquipes = qtdEquipes;
    }
    public int getQtdEquipe(){
        return qtdEquipes;
    }

    @Override
    public float bonus(){
        return qtdEquipes * 500;
    }
    @Override
    public String descricao(){
        return super.descricao() + ", Equipes: " + qtdEquipes + ", Bônus: " + bonus();
    }
}

File: lab4/C++/Gerente.cpp
#include "Gerente.hpp"

Gerente::Gerente(string nomeFuncionario, float salarioBase, int numero_de_equipes)
    : Funcionario(nomeFuncionario, salarioBase), numero_de_equipes(numero_de_equipes){}

float Gerente::bonusGerente()
{
    return numero_de_equipes * 500;
}
string Gerente::descricao()
{
    return "\nGerente: " + Funcionario::descricao() + ", Número de equipes: " + to_string(numero_de_equipes) + ", Bônus: " + to_string(bonusGerente());
}